src/codec/framed_read.rs

```diff
@@ -333,6 +90,256 @@ impl<T> FramedRead<T> {
     }
 }
 
+/// Decodes a frame.
+///
+/// This method is intentionally de-generified and outlined because it is very large.
+#[inline(never)]
+fn decode_frame(
+    hpack: &mut hpack::Decoder,
+    max_header_list_size: usize,
+    p: &mut Option<Partial>,
```

Comment:
Could this be called `partial`?

Reply:
Oh sorry. It would shadow… Hm hm...

Reply:
woops, yeah I meant to go back and give this a better name before submitting.  I did change it originally because of the shadowing issue.  Open to alternate name suggestions, or perhaps I should rename the other variable.